internal/provider/volume_resource.go

```diff
@@ -3,103 +3,105 @@ package provider
-	Appid      types.String `tfsdk:"app"`
-	Region     types.String `tfsdk:"region"`
-	Internalid types.String `tfsdk:"internalid"`
+	Id     types.String `tfsdk:"id"`
+	Name   types.String `tfsdk:"name"`
+	Size   types.Int64  `tfsdk:"size"`
+	Appid  types.String `tfsdk:"app"`
+	Region types.String `tfsdk:"region"`
 }
 
-func (t flyVolumeResourceType) GetSchema(context.Context) (tfsdk.Schema, diag.Diagnostics) {
-	return tfsdk.Schema{
+func (r *flyVolumeResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
 		MarkdownDescription: "Fly volume resource",
-		Attributes: map[string]tfsdk.Attribute{
-			"id": {
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
 				MarkdownDescription: "ID of volume",
-				Type:                types.StringType,
 				Computed:            true,
-				Optional:            true,
+				// Optional:            true,
 			},
-			"app": {
+			"app": schema.StringAttribute{
 				MarkdownDescription: "Name of app to attach to",
 				Required:            true,
-				Type:                types.StringType,
 			},
-			"size": {
+			"size": schema.Int64Attribute{
 				MarkdownDescription: "Size of volume in GB",
 				Required:            true,
-				Type:                types.Int64Type,
 			},
-			"name": {
+			"name": schema.StringAttribute{
 				MarkdownDescription: "name",
-				Type:                types.StringType,
 				Required:            true,
+				Validators: []validator.String{
+					stringvalidator.RegexMatches(
+						regexp.MustCompile(`^[a-z0-9_]+$`),
+						"only allows alphanumeric characters and underscores",
+					),
+				},
 			},
-			"region": {
+			"region": schema.StringAttribute{
 				MarkdownDescription: "region",
-				Type:                types.StringType,
 				Required:            true,
 			},
-			"internalid": {
-				MarkdownDescription: "Internal ID",
-				Type:                types.StringType,
-				Computed:            true,
-				Optional:            true,
-			},
 		},
-	}, nil
-}
-
-func (t flyVolumeResourceType) NewResource(ctx context.Context, in tfsdkprovider.Provider) (resource.Resource, diag.Diagnostics) {
-	provider, diags := convertProviderType(in)
-
-	return flyVolumeResource{
-		provider: provider,
-	}, diags
+	}
 }
 
-func (vr flyVolumeResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+func (r *flyVolumeResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
 	var data flyVolumeResourceData
 
 	diags := req.Plan.Get(ctx, &data)
 	resp.Diagnostics.Append(diags...)
 
-	q, err := graphql.CreateVolume(context.Background(), *vr.provider.client, data.Appid.Value, data.Name.Value, data.Region.Value, int(data.Size.Value))
+	q, err := graphql.CreateVolume(context.Background(), *r.client, data.Appid.ValueString(), data.Name.ValueString(), data.Region.ValueString(), int(data.Size.ValueInt64()))
```

Comment:
Like another one;

- Better to use the passed ctx
- Safer to return early if err != nil